src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -726,6 +733,8 @@ def from_date_range(
             dt_values = ns_values.series_to_datetime()
 
         dt_series = pd.Series(query_compiler=dt_values)
+        if remove_non_business_days:
+            dt_series = dt_series[dt_series.dt.dayofweek < 5]
```

Comment:
This seems to be removing just weekends, perhaps change the flag to `remove_non_business_days_excluding_holidays`? I'm not sure if there is a pandas function (outside of using the `CustomBusinessDay` param) that excludes holidays as well though so maybe this isn't necessary?

Reply:
pandas business day by default only means weekends but not holidays. We could revisit this if we decide to support CustomBusinessDay in the future IMO.